charts/kong/values.yaml

```diff
@@ -297,6 +297,8 @@ proxy:
   http:
     # Enable plaintext HTTP listen for the proxy
     enabled: true
+    serviceEnabled: true
+    containerEnabled: true
```

Comment:
I'd like to avoid yet more knobs if possible, and the interaction between four different sorts of enable toggles for Services feels unintuitive if you don't look at the template innards to see what each disables.

Do you see any issue with instead using special values for existing fields? You could set `servicePort="disabled"` or `containerPort="disabled"` instead of an integer to disable these piecemeal.

Alternately, these could be one of the less-commonly used options we list in the README but not in the base values.yaml, like `hostPort`.

Reply:
`Do you see any issue with instead using special values for existing fields? You could set servicePort="disabled" or containerPort="disabled" instead of an integer to disable these piecemeal.`

I feel it would be better to avoid mixing int & string value for same field to avoid any conversion error/issues. Instead, I'm thinking to use port value as 0(preferably) or 65536 to indicate the disabled state. Does it sound ok? 

As per this article https://ieeexplore.ieee.org/document/8816853, I see below about port "0"
` A special case is port 0 which is reserved but was never assigned. Traffic from and to port 0 is unusual, because it should not occur in the wild. As port 0 is unassigned, there is no common service listing for connections here. Furthermore, operating systems usually interpret the request to open port 0 as the request to allocate and open any currently unused port. Thus, traffic from and to port 0 should not occur, because no application should listen there and applications cannot send from port 0.`

Reply:
Yeah, special handling of `0` would be fine.

Reply:
updated the code with special handling of servicePort=0 to handle the use-case instead of adding extra knobs.